services/business/istio_validations.go

```diff
@@ -51,8 +51,14 @@ func (in *IstioValidationsService) GetNamespaceValidations(namespace string) (mo
 		return nil, err
 	}
 
+	pods, err := in.k8s.GetNamespacePods(namespace)
+	if err != nil {
+		log.Warningf("Cannot get pods for namespace %v.", namespace)
+		return nil, err
+	}
+
 	objectCheckers := []ObjectChecker{
-		checkers.RouteRuleChecker{istioDetails.RouteRules},
+		checkers.RouteRuleChecker{Namespace: namespace, PodList: pods.Items, RouteRules: istioDetails.RouteRules},
```

Comment:
I am not sure if this is correct.
For the GetServiceValidations, the pods are filtered by the service, or it should be if I read it correctly.
For the namespace, you are fetching all pods and this is checking against versions without matching per labels.
So, if I am not wrong we might have a bad-pod-1234 with labels version=v1, version=v2 and matches correctly.

The idea to add the RouteRuleChecker was to enable validations that are independent of the service, to not replicate in Istio Config what we are showing in Service Details.

Then, I would leave in GetNamespaceValidations validations that are per namespace, but not attached to the services, I think.

Perhaps, that means to split the RouteRuleChecker, or if PodList is not provided, to not use the pod checker on this.

Or, if we want to maintain the consistency, the pods should be narrowed/filtered per service in the evaluation.

If I didn't miss anything in the code, I couldn't review it entirely yet.

Reply:
> For the GetServiceValidations, the pods are filtered by the service, or it should be if I read it correctly.

You are right, they are filtered by app label (service level).

> For the namespace, you are fetching all pods and this is checking against versions without matching per labels.

You are right, for the GetNamespaceValidations I am not filtering per service. This solution then it could trigger some false positive. 

> The idea to add the RouteRuleChecker was to enable validations that are independent of the service, to not replicate in Istio Config what we are showing in Service Details.

I don't know if I fully follow you. I thought that we wanted to show same validations for each item within IstioConfig view. If we are using RouteRuleChecker, then it means that we will show weight/precedence validations as well.
IMO, it might be somewhat confusing to have some validations in one page but not to the other. 

> Or, if we want to maintain the consistency, the pods should be narrowed/filtered per service in the evaluation.

IMO, I would go this way. But perhaps I am missing something.

Reply:
No, it is good, I wanted to maintain consistency between both, that's the correct way, but I didn't want to pull all data related to services in the Istio Config logic.

I guess filtering pods ver labels "app x version" is what it is necessary.

I am also working actively in these files too and I was also thinking loud.